src/Sylius/Component/Core/Taxation/Applicator/OrderItemUnitsTaxesApplicator.php

```diff
@@ -57,18 +63,17 @@ public function __construct(CalculatorInterface $calculator, AdjustmentFactoryIn
     public function apply(OrderInterface $order, ZoneInterface $zone)
     {
         foreach ($order->getItems() as $item) {
-            $taxRate = $this->taxRateResolver->resolve($item->getVariant(), ['zone' => $zone]);
-            if (null === $taxRate) {
-                continue;
-            }
+            $taxRates = $this->taxRatesResolver->resolve($item->getVariant(), ['zone' => $zone]);
```

Comment:
It does not make much sense to apply multiple tax rate to a single order item unit. Not sure what use case you are trying to cover here, but I think this should always be a single tax rate.